packages/graph/hash_graph/lib/graph/src/store/postgres/query/mod.rs

```diff
@@ -0,0 +1,46 @@
+#![expect(dead_code, reason = "Work in progress")]
+
+//! Postgres implementation to compile queries.
+
+pub mod database;
+
+use std::fmt::{self, Formatter};
+
+use crate::store::{
+    postgres::query::database::{Column, Table},
+    query::QueryRecord,
+};
+
+/// A structural query, which can be compiled into a statement in Postgres.
+// TODO: Implement for `ReadQuery<DataType>`, `ReadQuery<PropertyType>`, etc. when associated types
+//       are implemented
+pub trait Query {
+    type Field: Field;
+    type Record: QueryRecord;
+
+    /// The table used for this `Query`.
+    fn base_table() -> Table;
+}
+
+/// A field used in queries.
```

Comment:
What is a "field"

Reply:
I'd prefer if you explain that slightly better in this type now anyway. I think it'll be too easy for us to forget to update some of this documentation (and as it might not appear in a diff we may miss it)

Reply:
I already updated the description slightly:
> /// An attribute of an ontology type or a knowledge element.

Reply:
I added a small change in 8a60ae771cd7e68ba60c6ae0978a70ce3e89d394 (#1210), so we can pick it up again later